packages/compiler-cli/src/ngtsc/typecheck/test/type_checker__get_symbol_of_template_node_spec.ts

```diff
@@ -67,6 +87,394 @@ runInEachFileSystem(() => {
+        expect(symbol.tsSymbol!.escapedName.toString()).toEqual('helloWorld');
+        expect(program.getTypeChecker().typeToString(symbol.tsType))
+            .toEqual('false | true | undefined');
+      });
+
+      it('should get a symbol for properties several levels deep', () => {
+        const fileName = absoluteFrom('/main.ts');
+        const templateString = `<div [inputA]="person.address.street"></div>`;
+        const {templateTypeChecker, program} = setup([
+          {
+            fileName,
+            templates: {'Cmp': templateString},
+            source: `
+              interface Address {
+                street: string;
+              }
+
+              interface Person {
+                address: Address;
+              }
+              export class Cmp {person?: Person;}
+            `,
+          },
+        ]);
+        const sf = getSourceFileOrError(program, fileName);
+        const cmp = getClass(sf, 'Cmp');
+        const nodes = getAstElements(templateTypeChecker, cmp);
+
+        const symbol =
+            templateTypeChecker.getSymbolOfNodeInComponentTemplate(nodes[0].inputs[0].value, cmp)!;
+        assertExpressionSymbol(symbol);
+        expect(symbol.tsSymbol!.escapedName.toString()).toEqual('street');
+        expect((symbol.tsSymbol!.declarations[0] as ts.PropertyDeclaration).parent.name!.getText())
+            .toEqual('Address');
+        expect(program.getTypeChecker().typeToString(symbol.tsType)).toEqual('string');
+      });
+
+      describe('should get symbols for conditionals', () => {
+        let templateTypeChecker: TemplateTypeChecker;
+        let cmp: ClassDeclaration<ts.ClassDeclaration>;
+        let program: ts.Program;
+        let templateString: string;
+
+        beforeEach(() => {
+          const fileName = absoluteFrom('/main.ts');
+          templateString = `
+        <div [inputA]="person?.address?.street"></div>
+        <div [inputA]="person ? person.address : noPersonError"></div>
+        <div [inputA]="person?.speak()"></div>
+      `;
+          const testValues = setup(
+              [
+                {
+                  fileName,
+                  templates: {'Cmp': templateString},
+                  source: `
+              interface Address {
+                street: string;
+              }
+
+              interface Person {
+                address: Address;
+                speak(): string;
+              }
+              export class Cmp {person?: Person; noPersonError = 'no person'}
+            `,
+                },
+              ],
+          );
+          templateTypeChecker = testValues.templateTypeChecker;
+          program = testValues.program;
+          const sf = getSourceFileOrError(program, fileName);
+          cmp = getClass(sf, 'Cmp');
+        });
+
+        it('safe property reads', () => {
+          const nodes = getAstElements(templateTypeChecker, cmp);
+          const safePropertyRead = nodes[0].inputs[0].value as ASTWithSource;
```

Comment:
I doubt this cast is necessary?

```suggestion
          const safePropertyRead = nodes[0].inputs[0].value;
```

There's some more cases in subsequent tests.